c++ training/PointersTraining/PointerExercises/S2E4/Source.cpp

```diff
@@ -0,0 +1,33 @@
+#include <iostream>
+
+/* Exercise 4 - Dynamic Allocation Dangers: Dangling Pointers
+There is another danger when using dynamic memory. Let�s see it:
+Allocate an int in the heap and assign it to an int pointer variable. Immediately after that, delete
+the allocated memory.
+We�ve previously discussed pointer validity. Check the validity of the pointer. If the pointer is
+valid, print �The pointer is valid!�. If it�s not, print �The pointer is not valid��.
+Run your program. You�ll see that the pointer is valid. Try accessing the value of whatever the
+pointer is pointing to and you�ll most likely crash. You�ve got a false positive in your validation
+due to a dangling pointer.
+Print to the console why you think this is happening. It�s important to keep in mind that after
+freeing memory in the heap, any pointer that was pointing to it should be set to null. From now
+on, this will be expected in the exercises.*/
+
+
+int main()
+{
+	int* SoonToBeADanglingPointer = new int(1);
+	delete SoonToBeADanglingPointer;
+
+	SoonToBeADanglingPointer = nullptr;
+
+	if (SoonToBeADanglingPointer)
+	{
+		std::cout << *SoonToBeADanglingPointer;
+	}
+
+	std::cout << "Well, this is simply happening cause even though the memory was set free again," << std::endl;
```

Comment:
This is not entirely true, if you run the program it will appear that it is valid, because although the value is deleted, the memory address still exists, it probably points to a garbage value or a strange value, but the fact that it points to something that is not 0 or null makes us return that it is valid. For this very reason it is a risk to have this type of situation

Reply:
I feel like that's what I said, I don't see what you say is not entirely true.

Reply:
In case this is because it has no trouble anymore, it's because I decided to upload the version modified with this exercise recommendation that we should set it to nullptr once we get rid of it.